src/vm.rs

```diff
@@ -415,6 +406,76 @@ impl TestContextObject {
     }
 }
 
+/// Provides additional methods for
+pub trait TraceItemDecorator {
```

Comment:
Do we need a trait here / will there be more than one impl?

Reply:
Unfortunately, Rust doesn't allow to implement associated functions to arrays. The only way to do it is to implement trait. But you are right, if we don't need multiple implementation, we can do the same different way. Let's just add needed functionality to `TraceAnalyzer`.

Reply:
Done.